contracts/HybridCustody.cdc

```diff
@@ -0,0 +1,533 @@
+import "CapabilityFactory"
+import "CapabilityProxy"
+import "CapabilityFilter"
+
+/*
+HybridCustody defines a framework for sharing accounts via account linking. In the contract,
+you will find three main resources being used:
+1. ChildAccount - A resource which maintains an AuthAccount Capability, and handles publishing
+    and revoking access of that account via another resource called a ProxyAccount
+2. ProxyAccount - A second resource which exists on the same accounta as the ChildAccount. The ProxyAccount
+    resource is the capability which is shared to a parent account. Each proxy has its own set of rules
+    designating what Capability types can be retrieved, and what underlying type the capability points to can be
+    given.
+3. Manager - A resource setup by the parent which manages all child accounts shared with it. The Manager resource
+    also maintains a set of accounts that it "owns", meaning it has a capability to the full ChildAccount resource and
+    would then also be able to manage the child account's links as it sees fit.
+*/
+pub contract HybridCustody {
+
+    pub let ChildStoragePath: StoragePath
+    pub let ChildPublicPath: PublicPath
+    pub let ChildPrivatePath: PrivatePath
+
+    pub let ManagerStoragePath: StoragePath
+    pub let ManagerPublicPath: PublicPath
+    pub let ManagerPrivatePath: PrivatePath
+
+    pub let LinkedAccountPrivatePath: PrivatePath
+    pub let BorrowableAccountPrivatePath: PrivatePath
+
+    // TODO: Events!
+
+    // An interface which gets shared to a Manager when it is given full ownership of an account.
+    pub resource interface Account {
+        pub fun getAddress(): Address
+        pub fun isChildOf(_ addr: Address): Bool
+        pub fun getParentsAddresses(): [Address]
+        pub fun borrowAccount(): &AuthAccount?
+    }
+
+    // A ChildAccount shares the BorrowableAccount capability to itelf with ProxyAccount resources
+    pub resource interface BorrowableAccount {
+        access(contract) fun borrowAccount(): &AuthAccount
+    }
+
+    // Public methods anyone can call on a child account
+    pub resource interface ChildAccountPublic {
+        pub fun getParentsAddresses(): [Address]
+        pub fun getParentStatuses(): {Address: Bool}
+        pub fun getRedeemedStatus(addr: Address): Bool?
```

Comment:
Can you add some comments explaining what a parent status and redeemed status is?